src/client/hooks/useLinkedData.js

```diff
@@ -1,25 +1,34 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useState, useReducer } from 'react'
 
 import { format } from '../../modules/3box-LD'
+import {
+  boxReducer,
+  initialState,
+  fetchedPublicProfileError,
+} from '../stateManagers/box'
 
-const useLinkedData = boxes => {
+const useLinkedData = () => {
   // const { connectedAccount } = useAragonApi()
   const connectedAccount = ''
   const [formattedProfile, setFormattedProfile] = useState({})
 
+  const [boxes, dispatch] = useReducer(boxReducer, initialState)
```

Comment:
Can you explain what's going on here?

Reply:
`useLinkedData` did not have its own access to reducer. Because I added exception checking to the function, I also added possibility to do `dispatch(fetchedPublicProfileError...`, and BoxWrapper can call useLinkedData without parameters, the function became self-sufficient...

```const BoxWrapper = ({ account, children, onSignatures, isViewMode }) => {
  const { boxes, dispatch } = use3Box(account, onSignatures)
  useLinkedData()
```